featuretools/tests/primitive_tests/test_feature_serialization.py

```diff
@@ -94,3 +107,132 @@ def serialize_name_unchanged(original):
 
     for feature_type in feature_type_list:
         serialize_name_unchanged(feature_type)
+
+
+@pytest.fixture
+def s3_client():
+    _environ = os.environ.copy()
+    from moto import mock_s3
+    with mock_s3():
+        s3 = boto3.resource('s3')
+        yield s3
+    os.environ.clear()
+    os.environ.update(_environ)
+
+
+@pytest.fixture
+def s3_bucket(s3_client):
+    s3_client.create_bucket(Bucket=BUCKET_NAME, ACL='public-read-write')
+    s3_bucket = s3_client.Bucket(BUCKET_NAME)
+    yield s3_bucket
+
+
+def test_serialize_features_mock_s3(es, s3_client, s3_bucket):
+    features_original = ft.dfs(target_entity='sessions', entityset=es, features_only=True)
+
+    ft.save_features(features_original, TEST_S3_URL)
+
+    obj = list(s3_bucket.objects.all())[0].key
+    s3_client.ObjectAcl(BUCKET_NAME, obj).put(ACL='public-read-write')
+
+    features_deserialized = ft.load_features(TEST_S3_URL)
+
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
```

Comment:
we can make this a helper function that tests in this file call to reduce code repetition